packages/actor-query-operation-leftjoin/lib/ActorQueryOperationLeftJoin.ts

```diff
@@ -36,12 +36,21 @@ export class ActorQueryOperationLeftJoin extends ActorQueryOperationTypedMediate
 
     // If the pattern contains an expression, filter the resulting stream
     if (operationOriginal.expression) {
+      const rightMetadata = await entries[1].output.metadata();
+      const expressionVariables = rightMetadata.variables;
       const config = { ...ActorQueryOperation.getAsyncExpressionContext(context) };
       const evaluator = new AsyncEvaluator(operationOriginal.expression, config);
       const bindingsStream = joined.bindingsStream
         .transform({
           autoStart: false,
           transform: async(bindings: Bindings, done: () => void, push: (item: Bindings) => void) => {
+            const optionalBindingsFulfilled = expressionVariables.every(variable => bindings.has(variable.value));
+
+            if (!optionalBindingsFulfilled) {
```

Comment:
Since the evaluation of this boolean is quite simple, I guess we can inline it into this if-statement.

Reply:
will do